packages/@lwc/template-compiler/src/parser/attribute.ts

```diff
@@ -252,3 +233,40 @@ function isTemplateDirective(attrName: string): boolean {
 export function attributeToPropertyName(attrName: string): string {
     return ATTRS_PROPS_TRANFORMS[attrName] || toPropertyName(attrName);
 }
+
+export class ParsedAttribute {
+    private readonly attributes: Map<string, IRAttribute> = new Map();
+
+    append(attr: IRAttribute): void {
+        this.attributes.set(attr.name, attr);
+    }
+
+    get(pattern: string | RegExp): IRAttribute | undefined {
+        const key = this._getKey(pattern);
+        if (key) {
+            return this.attributes.get(key);
+        }
+    }
+
+    pick(pattern: string | RegExp): IRAttribute | undefined {
+        const attr = this.get(pattern);
+        if (attr) {
+            this.attributes.delete(attr.name);
+        }
+        return attr;
+    }
+
+    private _getKey(pattern: string | RegExp): string | undefined {
```

Comment:
No needed to prefix it with `_` with TypeScript checking method visibility.

```suggestion
    private getKey(pattern: string | RegExp): string | undefined {
```